sway/tree/view.c

```diff
@@ -120,51 +140,26 @@ void view_autoconfigure(struct sway_view *view) {
 		height = view->swayc->height;
 		break;
 	case B_PIXEL:
-		if (view->swayc->layout > L_VERT
-				|| config->hide_edge_borders == E_NONE
-				|| config->hide_edge_borders == E_HORIZONTAL
-				|| (config->hide_edge_borders == E_SMART && other_views)) {
-			x = view->swayc->x + view->border_thickness;
-			width = view->swayc->width - view->border_thickness * 2;
-		} else {
-			x = view->swayc->x;
-			width = view->swayc->width;
-		}
-		if (view->swayc->layout > L_VERT
-				|| config->hide_edge_borders == E_NONE
-				|| config->hide_edge_borders == E_VERTICAL
-				|| (config->hide_edge_borders == E_SMART && other_views)) {
-			y = view->swayc->y + view->border_thickness;
-			height = view->swayc->height - view->border_thickness * 2;
-		} else {
-			y = view->swayc->y;
-			height = view->swayc->height;
-		}
+		x = view->swayc->x + view->border_thickness * view->border_left;
+		y = view->swayc->y + view->border_thickness * view->border_top;
+		width = view->swayc->width
+			- view->border_thickness * view->border_left
+			- view->border_thickness * view->border_right;
+		height = view->swayc->height
+			- view->border_thickness * view->border_top
+			- view->border_thickness * view->border_bottom;
 		break;
 	case B_NORMAL:
-		if (view->swayc->layout > L_VERT
-				|| config->hide_edge_borders == E_NONE
-				|| config->hide_edge_borders == E_HORIZONTAL
-				|| (config->hide_edge_borders == E_SMART && other_views)) {
-			x = view->swayc->x + view->border_thickness;
-			width = view->swayc->width - view->border_thickness * 2;
-		} else {
-			x = view->swayc->x;
-			width = view->swayc->width;
-		}
-		if (view->swayc->layout > L_VERT
-				|| config->hide_edge_borders == E_NONE
-				|| config->hide_edge_borders == E_VERTICAL
-				|| (config->hide_edge_borders == E_SMART && other_views)) {
-			// Height is: border + title height + border + view height + border
-			y = view->swayc->y + config->font_height
-				+ view->border_thickness * 2;
-			height = view->swayc->height - config->font_height
-				- view->border_thickness * 3;
-		} else {
-			y = view->swayc->y;
-			height = view->swayc->height;
-		}
+		// Height is: border + title height + border + view height + border
+		x = view->swayc->x + view->border_thickness * view->border_left;
+		y = view->swayc->y + config->font_height
```

Comment:
I want to clarify something: The title bar height is border + title height + border as per the comment. The single pixel bars above and below the title are *always* 1px, and overlap the border. So if your border thickness is 3px, you'll have 1px of the "border" colour, 2px of the remaining border thickness ("background" colour), then the font/texture height for the title (also "background" colour), then the opposite for below the title.

Your implementation culls the top 1px pixel bar if border_top is false, but calculates the view's y as if it were culling one entire border (3px in the above example).

I think the best option is to not cull anything at the top if using a normal border.

Reply:
If I understood your comment correctly, this should be fixed

Reply:
The calculation looks good now.

You also need to change the rendering code so it always renders the top pixel row when using border normal.

Reply:
Ok. I'll look at it again in a few hours after some sleep. I'm not seeing what I need to change at the moment. It looks like if border_top is false, then box.y would be con->y, which is the same y that the single pixel bar above the title is rendered at.

Reply:
Yeah, I think there's still some confusion. My suggestion is to always render the full title bar and top border regardless of any hide_edge_borders setting, which means `render_container_simple_border_normal` should not contain any references to border_top.

Reply:
Oh ok. Should be fixed now